src/lib/storage/table.cpp

```diff
@@ -18,56 +18,65 @@
 namespace opossum {
 
 Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+  _target_chunk_size = target_chunk_size;
+  _chunks.push_back(std::make_shared<Chunk>());
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  Assert(row_count() == 0, "Cannot add column to non-empty table");
+
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  _add_value_segment_to_chunk(_chunks[0], type);
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  auto last_chunk = _chunks[chunk_count() - 1];
+  if (last_chunk->size() == _target_chunk_size) {
+    last_chunk = std::make_shared<Chunk>();
+    for (auto column_type : _column_types) {
+      _add_value_segment_to_chunk(last_chunk, column_type);
+    }
+    _chunks.push_back(last_chunk);
+  }
+  last_chunk->append(values);
 }
 
-ColumnCount Table::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
-}
+ColumnCount Table::column_count() const { return _chunks[0]->column_count(); }
 
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  const auto completed_chunks_size = (chunk_count() - 1) * _target_chunk_size;
+  const auto last_chunk_size = _chunks[chunk_count() - 1]->size();
+  return completed_chunks_size + last_chunk_size;
 }
 
-ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  return ChunkID{0};
-}
+ChunkID Table::chunk_count() const { return ChunkID{_chunks.size()}; }
```

Comment:
Compiler Error: narrowing conversion of long unsigned int to uint32_t (unsinged int)
```suggestion
ChunkID Table::chunk_count() const { return ColumnCount{static_cast<ColumnCount::base_type>(_column_names.size())}; }
```